commcare_connect/opportunity/visit_import.py

```diff
@@ -311,3 +313,121 @@ def get_status_by_completed_work_id(dataset):
+            opportunity_access = {
+                oa.user.username: oa
+                for oa in OpportunityAccess.objects.filter(opportunity=opportunity).select_related("user")
+            }
+
+        for row in dataset:
+            row = list(row)
+            try:
+                latitude = Decimal(row[latitude_index])
+                longitude = Decimal(row[longitude_index])
+                radius = int(row[radius_index])
+                active = row[active_index].lower().strip() == "yes"
+                area_name = str(row[area_name_index])
+
+                if latitude < Decimal("-90") or latitude > Decimal("90"):
+                    raise ValueError("Latitude must be between -90 and 90 degrees")
+
+                if longitude < Decimal("-180") or longitude > Decimal("180"):
+                    raise ValueError("Longitude must be between -180 and 180 degrees")
+
+                if not radius:
+                    raise ValueError("Radius must be an integer")
+
+                if active_index is None or row[active_index].lower().strip() not in ["yes", "no"]:
+                    raise ValueError("Active status must be 'yes' or 'no'")
+
+                if area_name_index is None or not isinstance(row[area_name_index], str):
+                    raise ValueError("Area name is not valid.")
+
+                if not username_index or not row[username_index]:
+                    catchment = CatchmentArea(
+                        latitude=latitude,
+                        longitude=longitude,
+                        radius=radius,
+                        opportunity=opportunity,
+                        name=area_name,
+                        active=active,
+                    )
+                    missing_catchments += 1
+                    to_create.append(catchment)
+                elif row[username_index] in opportunity_access:
+                    username = row[username_index]
+                    catchment = None
+                    created = None
+                    if CATCHMENT_ID in headers and row[_get_header_index(headers, CATCHMENT_ID)]:
+                        catchment_id = row[_get_header_index(headers, CATCHMENT_ID)]
+                        catchment, created = CatchmentArea.objects.get_or_create(
+                            id=catchment_id,
+                            defaults={
+                                "latitude": latitude,
+                                "longitude": longitude,
+                                "radius": radius,
+                                "opportunity_access": opportunity_access[username],
+                                "opportunity": opportunity,
+                                "name": area_name,
+                                "active": active,
+                            },
+                        )
+
+                    if not created:
+                        catchment.latitude = latitude
```

Comment:
`catchment` can be `None` here. what happens in that case?

Reply:
Added this case too.